app/src/main/java/org/mifos/mobilebanking/presenters/BeneficiaryApplicationPresenter.java

```diff
@@ -78,6 +80,7 @@ public void onError(Throwable e) {
                     public void onNext(BeneficiaryTemplate beneficiaryTemplate) {
                         getMvpView().hideProgress();
                         getMvpView().showBeneficiaryTemplate(beneficiaryTemplate);
+                        getMvpView().setVisibility(View.VISIBLE);
```

Comment:
Add this inside `onComplete()`

Reply:
Remove this from `onNext()`

Reply:
@miPlodder I now see what you mean, I am using the methods innocrectly, onNext() does work to prepare the UI, and onComplete() finishes the process